mobile/src/feeds/home-feeds-screen.tsx

```diff
@@ -1,28 +1,44 @@
 import React from 'react';
-import { FlatList, Text, View} from 'react-native';
+import { FlatList, Text, View, StyleSheet } from 'react-native';
 import { TopNavBar } from '../topNavBar/top-nav-bar';
 
-const navBar =  new TopNavBar('Feeds');
-const dataList = [
-{key: 'First Feed'},
-{key: 'Second Feed'},
-{key: 'Third Feed'},
-{key: 'Fourth Feed'},
-{key: 'Fifth Feed'},
-{key: 'Sixth Feed'},
-{key: 'Seventh Feed'},
-{key: 'Ten'}
-];
+import { Feed } from '../reducers/organization/feed.model';
+import { connect } from 'react-redux';
+import { AppState } from '../reducers/app.reducer';
 
+import { FeedListItem } from './feed';
 
-export class HomeFeedsScreen extends React.Component {
+import { screenStyles as styles } from './styles';
+
+const navBar = new TopNavBar('Feeds');
+
+interface FeedsScreenProps {
+    feeds: Feed[];
+}
+
+const mapStateToProps = (state: AppState): FeedsScreenProps => ({
+    feeds: state.organization.feeds.map(feed => {
+        feed.employee = state.organization.employees.employeesById.get(feed.employeeId);
```

Comment:
this is just wrong, you change a part of the immutable state outside of the reducer. Even more, you do it in *map* function, which is supposed to be pure by definition. If you need employees in props, just map them directly. You should have a render function call when this array change.